qulice-pmd/src/main/java/com/qulice/pmd/PmdValidator.java

```diff
@@ -45,46 +46,53 @@
  * @version $Id$
  * @since 0.3
  */
-public final class PmdValidator implements Validator {
+public final class PmdValidator implements ResourceValidator {
+
+    /**
+     * Environment to use.
+     */
+    private final transient Environment env;
+
+    /**
+     * Constructor.
+     * @param env Environment to use.
+     */
+    public PmdValidator(final Environment env) {
+        this.env = env;
+    }
 
     @Override
-    public void validate(final Environment env) throws ValidationException {
-        final SourceValidator validator = new SourceValidator(env);
-        final Collection<DataSource> sources = this.getSources(env);
-        final File base = env.basedir();
-        final String path = base.getPath();
-        validator.validate(sources, path);
-        final Collection<RuleViolation> violations = validator.getViolations();
-        final int size = violations.size();
-        if (!violations.isEmpty()) {
-            throw new ValidationException(
-                "%d PMD violations (see log above)",
-                size
-            );
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public Collection<Violation> validate(final Collection<File> files) {
+        final SourceValidator validator = new SourceValidator(this.env);
+        final Collection<DataSource> sources = new LinkedList<>();
+        for (final File file : files) {
+            sources.add(new FileDataSource(file));
         }
-        Logger.info(
-            this,
-            "No PMD violations found in %d files",
-            sources.size()
+        final Collection<RuleViolation> breaches = validator.validate(
+            sources, this.env.basedir().getPath()
         );
+        final Collection<Violation> violations = new LinkedList<>();
```

Comment:
@krzyk can we use more performant `ArrayList` here with initial size set to `breaches.size()`?